src/sidebar/util/thread-annotations.js

```diff
@@ -0,0 +1,75 @@
+import buildThread from './build-thread';
+import memoize from './memoize';
+import * as metadata from './annotation-metadata';
+import { generateFacetedFilter } from './search-filter';
+import filterAnnotations from './view-filter';
+import { shouldShowInTab } from './tabs';
+
+/** @typedef {import('./build-thread').Thread} Thread */
+/** @typedef {import('./build-thread').Options} BuildThreadOptions */
+/** @typedef {import('../store/modules/selection').ThreadState} ThreadState */
```

Comment:
You can combine the typedefs into a single `/** ... */` comment here. It won't save any lines in this case, but I'm just noting for future reference that you don't have to use a separate `/** ... */` for each import.